src/main/java/software/amazon/event/kafkaconnector/batch/BestEffortInOrderEventBridgeBatching.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.batch;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Stream;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+public class BestEffortInOrderEventBridgeBatching implements EventBridgeBatchingStrategy {
+
+  private final Collector<
+          MappedSinkRecord<PutEventsRequestEntry>,
+          PutEventsRequestEntrySizeCollector,
+          Stream<List<MappedSinkRecord<PutEventsRequestEntry>>>>
+      collector =
+          Collector.of(
+              PutEventsRequestEntrySizeCollector::new,
+              PutEventsRequestEntrySizeCollector::accumulate,
+              (left, right) -> {
+                throw new IllegalStateException();
```

Comment:
We need to document all those exceptions and verify we handle them accordingly in the caller

Reply:
You're right, documentation would be good. The exception could only happen if the provided stream of records is parallel. It's checked when apply the strategy:
```
  @Override
  public Stream<List<MappedSinkRecord<PutEventsRequestEntry>>> apply(
      Stream<MappedSinkRecord<PutEventsRequestEntry>> records) {
    if (records.isParallel()) {
      throw new IllegalArgumentException("Stream must not be parallel.");
    }
    return records.collect(collector);
  }
```
It will never happen in your usage since we call the `apply` method with a non-parallel stream.

Reply:
Agree, just to be sure: should we make those checked exceptions then, indicated with `throws` or just doc it?

Reply:
Documentation is enough IMO. The call of the method is from `EventBridgeWriter` which is under 'our' control.